### PR TITLE
Add Openshift provider type option to the module

### PR DESCRIPTION
**Pull request by dkorn**
The choices are: 
- 'openshift-origin'
- 'openshift-enterprise'

### CODE REVIEW

**Comment by dkorn** [comment]
@enoodle @cben @simon3z please review

File: library/manageiq.py [hunk 1]
```diff
@@ -178,13 +187,14 @@ def main():
...
     provider_name = module.params['name']
+    provider_type = module.params['type']
     hostname      = module.params['hostname']
     port          = module.params['port']
     token         = module.params['token']
 
     manageiq = ManageIQ(module, url, username, password)
 
-    res_args = manageiq.add_or_update_provider(provider_name, hostname, port, token)
+    res_args = manageiq.add_or_update_provider(provider_name, provider_type, hostname, port, token)
```
Comment (cben) [inline library/manageiq.py:197]: it may be time to start passing these k=v style

**Comment by cben** [comment]
I think `type` should take `['openshift_origin', 'openshift_enterprise']`.
While you only care about openshift providers at this stage, in the future adding say Amazon, oVirt, etc. providers makes equal sense...

File: library/manageiq.py (hunk 1 above)
Comment (dkorn) [reply library/manageiq.py:197]: @cben why would that be useful? `add_or_update_provider` doesn't have default values

**Comment by dkorn** [comment]
> I think type should take ['openshift_origin', 'openshift_enterprise'].
> While you only care about openshift providers at this stage, in the future adding say Amazon, oVirt, etc. providers makes equal sense...

I agree, done

**Comment by cben** [comment]
LGTM

**Comment by enoodle** [comment]
LGTM :+1:

**Pull request merged**